xboxlib.c

```diff
@@ -204,6 +205,31 @@ void gigimport(giant g, unsigned char *buff,int len){
 	
 
 }
+#endif
+
+void gigimport(giant g, unsigned char *buff, int len) {
+
+	// copy buffered 'number' into giant's number buffer
+	//int count;
+	memcpy(g->n, buff, len);
+
+	// the giants number buffer stores large numbers as
+	// multiple 16bit numbers, so if the length of our buffer is odd
+	// our number wont fit properly.
+	// to fix this we add a trailing 0x00 byte
+	if (len & 1)
```

Comment:
Wouldn't this case means errors on different endianess? We shouldn't just memcpy an odd number of 8 bit words into a 16 bit word location, then clear one of the halfs of the 16 bit words.

If this case is not used in the source, I'd recommend to `assert((len % 2) == 0)` instead.

If it *is* used, then the type of `buff` should probably be fixed and the overall idea of the code reviewed. This wouldn't have to happen in this PR though, but an issue should be created.

Reply:
In the source I see that gigimport is always called with a len argument of 256 or 4, both of which are even numbers so, in practice, the odd case never happens. I would opt for the assert solution then.